WordPress/src/main/java/org/wordpress/android/ui/posts/PostUploadService.java

```diff
@@ -88,15 +90,27 @@ public void onCreate() {
     }
 
     @Override
-    public void onStart(Intent intent, int startId) {
+    public void onDestroy() {
+        super.onDestroy();
+        // Cancel current task, it will reset post from "uploading" to "local draft"
+        if (mCurrentTask != null) {
```

Comment:
It can be null if mPostsList is empty (we won't instantiate a new task in `uploadNextPost`). It shouldn't be empty on first call.